Sources/PexipRTC/Internal/WebRTCMediaConnection.swift

```diff
@@ -105,20 +105,32 @@ final class WebRTCMediaConnection: NSObject, MediaConnection {
 
     // MARK: - MediaConnection (tracks)
 
-    func setMainAudioTrack(_ audioTrack: LocalAudioTrack?) {
+    func setMainAudioTrack(_ audioTrack: LocalAudioTrack?) throws {
         mainLocalAudioTrack = audioTrack.valueOrNil(WebRTCLocalAudioTrack.self)
-        mainAudioTransceiver = mainAudioTransceiver ?? connection.addTransceiver(of: .audio)
-        mainAudioTransceiver?.sender.track = mainLocalAudioTrack?.rtcTrack
-        mainLocalAudioTrack?.capturingStatus.$isCapturing.sink { [weak self] isCapturing in
+
+        if mainAudioTransceiver == nil && mainLocalAudioTrack != nil {
+            mainAudioTransceiver = connection.addAudioTransceiver(.sendOnly)
+        }
+
+        try mainAudioTransceiver?.send(from: mainLocalAudioTrack?.rtcTrack)
+
+        let status = mainLocalAudioTrack?.capturingStatus ?? CapturingStatus(isCapturing: false)
+        status.$isCapturing.sink { [weak self] isCapturing in
             self?.muteAudio(!isCapturing)
         }.store(in: &cancellables)
```

Comment:
Is it cancelled when the track changes to `nil`? Or does it stay active in `cancellables`?

Reply:
That's a good point, the cancellable will stay active. It's better to have separate cancellables for audio and video tracks to make sure they are cancelled when deinitialized.